### PR TITLE
Scale based layer visibility cosmetics

### PR DESCRIPTION
**Pull request by pvalsecc**
Made layers/labels visibility more consistent.
Make out of range layers greyed out in the tree.
Add a menu entry to zoom to a layer's scale range.

### CODE REVIEW

File: src/core/layertree/qgslayertreemodel.h [hunk 1]
```diff
@@ -254,6 +254,7 @@ class CORE_EXPORT QgsLayerTreeModel : public QAbstractItemModel
 
     //! emit dataChanged() for layer tree node items
     void recursivelyEmitDataChanged( const QModelIndex& index = QModelIndex() );
+    void refreshScaleBasedLayers( const QModelIndex& index = QModelIndex() );
```
Comment (nyalldawson) [inline src/core/layertree/qgslayertreemodel.h]: Can you add a doxygen string here too?

**Comment by nyalldawson** [comment]
Looks good to me - nice stuff! Can you add some unit tests for QgsMapLayer::isInScaleRange to protect against regressions?

**Comment by nyalldawson** [comment]
Oh - one last minor thing. If you rename the final commit to "[FEATURE] Add a menu entry to zoom to a layer's scale range" it will automatically create a issue for the documentation team to add this to the QGIS manual.

**Comment by pvalsecc** [comment]
OK, all comments resolved.

**Comment by nyalldawson** [comment]
Thank you!

**Pull request merged**